src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateItemEventHandler.java

```diff
@@ -161,6 +170,27 @@ public UpdateItemEventHandler(Storage storage, MappingMetadataCache mappingMetad
     return future;
   }
 
+  private Future<DataImportEventPayload> addHoldingToPayloadIfNeeded(DataImportEventPayload dataImportEventPayload, Context context, Item updatedItem) {
+    Promise<DataImportEventPayload> promise = Promise.promise();
+    if (dataImportEventPayload.getContext().get(HOLDINGS.value()) == null || StringUtils.isBlank(dataImportEventPayload.getContext().get(HOLDINGS.value()))) {
+      HoldingsRecordCollection holdingsRecordCollection = storage.getHoldingsRecordCollection(context);
+      holdingsRecordCollection.findById(updatedItem.getHoldingId(),
+        success -> {
+          LOG.info("Successfully retrieved Holdings for the hotlink by id: {}", updatedItem.getHoldingId());
+          dataImportEventPayload.getContext().put(HOLDINGS.value(), Json.encodePrettily(success.getResult()));
+          promise.complete(dataImportEventPayload);
+        },
+        failure -> {
+          LOG.warn(format("Error retrieving Holdings for the hotlink by id %s cause %s, status code %s", updatedItem.getHoldingId(), failure.getReason(), failure.getStatusCode()));
+          promise.complete(dataImportEventPayload);
+        });
+    } else {
+      LOG.info("Holdings already exists in payload with for the hotlink");
```

Comment:
I think we can include holdings id to this log from updatedItem.getHoldingId()
and possibly make this log as debug but its up to you